### PR TITLE
Add support for specifying --service-name when doing an AD join

### PR DESCRIPTION
**Pull request by alavaliant**
This merge request adds the option to specify the ad_join_service_names parameter  which passes the given SPN values to adcli's --service-name option.

This allows extra kerberos SPN values to be automatically added to the system keytab when a host is joined to AD.

Have tested and confirmed that if the new parameter isn't given then the default adcli command doesn't get altered.

### CODE REVIEW

**Comment by acjohnson** [comment]
Apologies I just noticed this PR thanks to #10 coming through today lol

File: manifests/join.pp [hunk 1]
```diff
@@ -33,6 +34,11 @@
     } else {
       $ad_join_domain_controller_command = ''
     }
+    if $ad_join_service_names != [] {
+        $ad_join_service_names_comand = join([" --service-name='", join($ad_join_service_names, "' --service-name='"), "'"], '')
```
Comment (acjohnson) [inline manifests/join.pp]: please correct and spell command properly

File: manifests/join.pp [hunk 2]
```diff
@@ -33,6 +34,11 @@
...
+    } else {
+      $ad_join_service_names_comand = ''
```
Comment (acjohnson) [inline manifests/join.pp]: please correct and spell command properly

File: manifests/join.pp [hunk 3]
```diff
@@ -51,7 +57,7 @@
     exec {'adcli_join':
       command   => "/bin/echo -n \'${ad_join_password}\' | /usr/sbin/adcli join ${ad_join_domain_controller_command} --login-user=\'${ad_join_username}\' \
 --domain=\'${ad_domain}\' --domain-ou=\'${ad_join_ou}\' --stdin-password --verbose ${ad_join_os_command} \
-${ad_join_os_version_command} ${ad_join_os_service_pack_command}",
+${ad_join_os_version_command} ${ad_join_os_service_pack_command} ${ad_join_service_names_comand}",
```
Comment (acjohnson) [inline manifests/join.pp]: please correct and spell command properly

File: manifests/join.pp (hunk 1 above)
Comment (alavaliant) [reply manifests/join.pp]: comand has been corrected to command

File: manifests/join.pp (hunk 3 above)
Comment (alavaliant) [reply manifests/join.pp]: comand has been corrected to command

File: manifests/join.pp (hunk 2 above)
Comment (alavaliant) [reply manifests/join.pp]: comand has been corrected to command

**Comment by alavaliant** [comment]
Thanks for reviewing.    Have made requested correction and fixed up merge conflicts with master due to #10 being merged.     So hopefully should be ready for merging now.

**Comment by alavaliant** [comment]
@acjohnson  Is there anything else I need to do with this merge request to get it accepted?

**Pull request merged**